whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java

```diff
@@ -183,11 +186,23 @@ private List<String> extractRecipients(final DeliveryStatus deliveryStatus) {
                 LOGGER.error("Wrong formatted recipient {}", recipient);
                 continue;
             }
-            recipients.add(finalRecipientMatcher.group(2));
+            final String email = finalRecipientMatcher.group(2);
+            if (isValidEmail(email)) {
+                recipients.add(email);
```

Comment:
Double-check the impact of an empty recipient in parser response (if the email is not valid)

Reply:
Checked code, it's OK if no email address in `parse()` response, message will still be deleted.

However please add a test for this (i.e. parse a bounce response with-too long email).

Reply:
I added a test called 
permanent_delivery_failure_to_one_recipient_too_long_address_then_delete_message_without_marking_undeliverable_address
in MailBounceTestIntegration
can you double check? This should take care that the message is not still there and that the email is not stored in database